x-pack/plugins/apm/server/lib/services/get_services/get_services_items.ts

```diff
@@ -19,72 +20,315 @@ import {
+                },
+              },
+            ],
+          },
+        },
+        aggs: {
+          services: {
+            terms: {
+              ...projection.body.aggs.services.terms,
+              size: MAX_NUMBER_OF_SERVICES,
             },
-            events: {
-              terms: { field: PROCESSOR_EVENT },
+            aggs: {
+              agent_name: {
+                top_hits: {
+                  _source: [AGENT_NAME],
+                  size: 1,
+                },
+              },
             },
-            environments: {
-              terms: { field: SERVICE_ENVIRONMENT },
+          },
+        },
+        size: 0,
+      },
+    })
+  );
+
+  const { aggregations } = response;
+
+  if (!aggregations) {
+    return [];
+  }
+
+  return aggregations.services.buckets.map((bucket) => ({
+    name: bucket.key as string,
+    value: (bucket.agent_name.hits.hits[0]?._source as {
+      agent: {
+        name: string;
+      };
+    }).agent.name,
+  }));
+};
+
+const getTransactionRate = async ({ setup, projection }: AggregationParams) => {
+  const response = await setup.client.search(
+    mergeProjection(projection, {
+      body: {
+        size: 0,
+        query: {
+          bool: {
+            filter: [
+              ...projection.body.query.bool.filter,
+              {
+                term: {
+                  [PROCESSOR_EVENT]: 'transaction',
+                },
+              },
+            ],
+          },
+        },
+        aggs: {
+          services: {
+            terms: {
+              ...projection.body.aggs.services.terms,
+              size: MAX_NUMBER_OF_SERVICES,
             },
           },
         },
       },
-    },
+    })
+  );
+
+  const { aggregations } = response;
+
+  if (!aggregations) {
+    return [];
+  }
+
+  const deltaAsMinutes = (setup.end - setup.start) / 1000 / 60;
+
+  return arrayUnionToCallable(aggregations.services.buckets).map((bucket) => {
+    const transactionsPerMinute = bucket.doc_count / deltaAsMinutes;
+    return {
+      name: bucket.key as string,
+      value: transactionsPerMinute,
+    };
   });
+};
+
+const getErrorRate = async ({ setup, projection }: AggregationParams) => {
+  const response = await setup.client.search(
+    mergeProjection(projection, {
+      body: {
+        size: 0,
+        query: {
+          bool: {
+            filter: [
+              ...projection.body.query.bool.filter,
+              {
+                term: {
+                  [PROCESSOR_EVENT]: 'error',
+                },
+              },
+            ],
+          },
+        },
+      },
+    })
+  );
+
+  const { aggregations } = response;
 
-  const resp = await client.search(params);
-  const aggs = resp.aggregations;
+  if (!aggregations) {
+    return [];
+  }
 
-  const serviceBuckets = aggs?.services.buckets || [];
+  const deltaAsMinutes = (setup.end - setup.start) / 1000 / 60;
```

Comment:
Maybe you could extract `deltaAsMinutes` to a function and use it on `getTransactionRate` too.